### PR TITLE
Add support for changing counters and subtitle

### PR DESCRIPTION
**Pull request by piotrpasich**
## Hey, I just made a Pull Request!

Adding new functionality to the CatalogTable to make it possible to change the counters and prepare subtitles. 

```
<CatalogTable columns={columns} actions={actions} calculateCounterFunction={getEntityCounterForDefault} buildSubtitleFunction={getEntityCounterSubtitle}/>
```

where `getEntityCounterSubtitle` and `getEntityCounterForDefault` have array Entities[] in parameters

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog/src/components/CatalogTable/CatalogTable.tsx [hunk 1]
```diff
@@ -61,6 +61,8 @@ export interface CatalogTableProps {
   tableOptions?: TableProps<CatalogTableRow>['options'];
   emptyContent?: ReactNode;
   subtitle?: string;
+  calculateCounterFunction?: Function,
+  buildSubtitleFunction?: Function
```
Comment (camilaibs) [inline plugins/catalog/src/components/CatalogTable/CatalogTable.tsx:65]: I would recommend using a more specific type so we know what these functions accept and return:
```suggestion
  calculateCounterFunction?: (entities: Entity[]) => number,
  buildSubtitleFunction?: (entities: Entity[]) => string
```

**Review by camilaibs — CHANGES_REQUESTED** [review]
Thank you for contributing, left a initial comment:

**Comment by camilaibs** [comment]
> ## Missing Changesets
> The following package(s) are changed by this PR but do not have a changeset:
> 
> * **@backstage/plugin-catalog**
> 
> See [CONTRIBUTING.md](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets) for more information about how to add changesets.
> 
> ## Changed Packages
> Package Name	Package Path	Changeset Bump	Current Version
> @backstage/plugin-catalog	plugins/catalog	**none**	`v1.23.0-next.2`

@piotrpasich could you generate the changesets, please?

File: plugins/catalog/src/components/CatalogTable/CatalogTable.tsx (hunk 1 above)
Comment (vinzscam) [reply plugins/catalog/src/components/CatalogTable/CatalogTable.tsx:65]: I will iterate with something like:

```
  calculateCounterFunction?: (options: { context: EntityListContextProps }) => number,
  buildSubtitleFunction?: (options: { context: EntityListContextProps }) => string
```

Keep in mind that if pagination is enabled, `entities` will contain only a subset of all entities present in the catalog. Therefore, you might need to consider additional factors when building your logic.

**Pull request closed without merging**